tests/compile-test.rs

```diff
@@ -23,3 +23,23 @@ fn run_mode(mode: &'static str) {
 fn compile_test() {
     run_mode("compile-fail");
 }
+
+#[test]
+fn dogfood() {
```

Comment:
I'm not sure what you mean with deleted build deps, All this test does is build clippy as a binary and then execute it, it doesn't even build dependencies, but instead takes those that were built for clippy.